src/dmd/ctfeexpr.d

```diff
@@ -328,7 +328,6 @@ extern (C++) UnionExp copyLiteral(Expression e)
     if (e.op == TOKarrayliteral)
     {
         auto ale = cast(ArrayLiteralExp)e;
-        auto basis = ale.basis ? copyLiteral(ale.basis).copy() : null;
         auto elements = copyLiteralArray(ale.elements, ale.basis);
```

Comment:
I'm suspicious that the `basis` was intended to be used on line 331 instead of `ale.basis`. This is a serious issue.

Reply:
Introducing PR was #5085. I believe your concerns are unfounded based on the change there (it's always been dead).

Reply:
It looks to me like the bug was Kenji forgetting to use `basis` in the next line. The code sure looks like it was intended to. The fact that it did not trigger a failure of the test suite doesn't mean it's not a bug. I bet if you make the change to use it in the next like, it will also pass the test suite.

I cannot approve this PR with that line deleted without some investigation into what was the intended use of it. If you don't want to investigate it, and move ahead with this PR, I suggest flagging it with a comment and leave the line as is, and I'll approve it.

Reply:
@wilzbach issues like this one are why having a warning on unused variables would be problematical.

Reply:
I'm just saying let's not discount an early version of the PR needed it, but later changes made it irrelevant.

As for warnings such as -Wunused, it's only useful for new PRs, particularly refactorings.

Reply:
OK, I re-reviewed #5085, and this line is not hiding a bug, it genuinely is dead.

Reference, implementation of `copyArrayLiteral()`
https://github.com/dlang/dmd/pull/5085/files#diff-a86b7363d77bfa5ce809fa1d3985a086R309

https://github.com/dlang/dmd/blob/716303a41a642fb3ce2c8d0d95b0a7df4047713a/src/dmd/ctfeexpr.d#L299-L305

The call to `copyLiteral().copy()` was moved here.

Reply:
Thanks for doing the research.